language/move-stdlib/tests/VectorTests.move

```diff
@@ -0,0 +1,398 @@
+#[test_only]
+module 0x1::VectorTests {
+    use 0x1::Vector as V;
+
+    struct R has store { }
```

Comment:
This is sick, and I hadn't really considered it. I think it pushes me a bit in the camp that maybe we shouldn't have the separate sort of "extension module for tests" thing we were talking about. 

I think it might be more clean to just have a separate module. And do `#[test_only]` like we do with `#[cfg(test)]` in Rust (all in the same module decl)